tests/pipe.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+var test = require('tape');
+
+var Buffer = require('buffer').Buffer;
+var nock = require('nock');
+var PassThrough = require('readable-stream').PassThrough;
+var request = require('request');
+
+var MockRequest = require('../index.js').Request;
+var MockResponse = require('../index.js').Response;
+
+test('nock should return response', function t(assert) {
+
+    nock('http://localhost:5555')
+        .post('/derp')
+        .reply(200, '{a');
+
+    request.post('http://localhost:5555/derp', onResponse);
+
+    function onResponse(err, res, body) {
+        assert.ifError(err, 'Unexpected error in resonse');
+        assert.strictEqual('{a', String(body));
+        assert.end();
+    }
+});
```

Comment:
I'm not sure why the above test is here as it tests nock and request.

Reply:
The point here was to prove that these modules actually work... What I wanted was a reproduction of a legitimate proxy case. However, when you have 3 different modules piped together and all you're looking at is what comes out of the pipe chain, it's not obvious which thing is broken. So really I just saw this as pre-requisite to the real proxy test through the hammock req/res pair.

Reply:
that makes sense. 💯  I suggest we move this into a test that is separate from the unit test.